### ISSUE TITLE
search_releases() returns wrong info

### ISSUE THREAD

**Issue by lazka** [alastair/python-musicbrainzngs#194]
The following script:

```
import musicbrainzngs
import pprint

musicbrainzngs.set_useragent("foo", "0.1")
releases = musicbrainzngs.search_releases('"Colour Of Snow" AND artist:"Polarkreis 18" AND tracks:11')
pprint.pprint(releases)
```

gives

```
{'release-count': 1,
 'release-list': [{'artist-credit': [{'artist': {'id': '90b18d97-718b-4a95-982d-b14019d084c0',
                                                 'name': 'Polarkreis 18',
                                                 'sort-name': 'Polarkreis 18'}}],
                   'artist-credit-phrase': 'Polarkreis 18',
                   'asin': 'B001EL2CBY',
                   'barcode': '0602517853607',
                   'country': 'DE',
                   'date': '2008-10-17',
                   'ext:score': '100',
                   'id': '87c070fc-90d5-39d1-b0ca-777236588378',
                   'label-info-list': [{'catalog-number': '1785360',
                                        'label': {'id': 'bb7f25ac-2da7-4484-ae39-8db7df9a6c17',
                                                  'name': 'Vertigo Berlin'}}],
                   'medium-count': 1,
                   'medium-list': [{},
                                   {'disc-count': 1,
                                    'disc-list': [],
                                    'format': 'CD',
                                    'track-count': 11,
                                    'track-list': []}],
                   'release-event-list': [{'area': {'id': '85752fda-13c4-31a3-bee5-0e5cb1f51dad',
                                                    'iso-3166-1-code-list': ['DE'],
                                                    'name': 'Germany',
                                                    'sort-name': 'Germany'},
                                           'date': '2008-10-17'}],
                   'release-group': {'id': '7e10bfaa-7bf1-3c76-844e-c5b4f13fd374',
                                     'primary-type': 'Album',
                                     'type': 'Album'},
                   'status': 'Official',
                   'text-representation': {'language': 'eng',
                                           'script': 'Latn'},
                   'title': 'The Colour of Snow'}]}
```

https://musicbrainz.org/release/87c070fc-90d5-39d1-b0ca-777236588378

1) track-count is returned as 11 but the release has only 10 tracks.
2) medium-list contains an empty dict

**Comment by lazka** [comment]
ok, as pointed out on IRC I missed the "0" track in the release :/

**Comment by alastair** [comment]
The problem with the `medium-list` is because the xml is
```
<medium-list count="1">
   <track-count>11</track-count>
   <medium>
      <format>CD</format>
      <disc-list count="1"/>
      <track-list count="11"/>
   </medium>
</medium-list>
```

and we parse it in `parse_medium_list` as 
```
return [parse_medium(m) for m in ml]
```
except `<track-count>` isn't a medium.

the [mmd](https://github.com/metabrainz/mmd-schema/blob/0df6c254fbb86113f06ae92ef78fa6fc7af60c1d/schema/musicbrainz_mmd-2.0.rng#L2014-L2026) shows that `track-count` is optional and then there are 1 or more `medium` elements

**Comment by lazka** [comment]
One more:

```
import musicbrainzngs
import pprint

musicbrainzngs.set_useragent("foo", "0.1")

release = musicbrainzngs.get_release_by_id("87c070fc-90d5-39d1-b0ca-777236588378", includes=["recordings", "artists", "artist-credits"])
pprint.pprint(release)
```

The returned release is missing the first track and says it contains 10 tracks.

**Comment by mineo** [comment]
Track #0 on that one is a pregap track.

**Comment by alastair** [comment]
there's a `<pregap>` element in the xml which you're not getting: http://musicbrainz.org/ws/2/release/87c070fc-90d5-39d1-b0ca-777236588378?inc=recordings

This is fixed and will be in 0.6; https://github.com/alastair/python-musicbrainzngs/pull/165

**Comment by alastair** [comment]
I'm not sure where to put the `<track-count>` element. This only shows in search results, not in entity requests.
I think we should put it at the same level as `medium-list` as a `medium-track-count` key.
